src/steamship/data/plugin/prompt_generation_plugin_instance.py

```diff
@@ -0,0 +1,105 @@
+from steamship.base.client import Client
+from steamship.data.plugin.plugin_instance import CreatePluginInstanceRequest
+from steamship.data.tags.tag_constants import TagKind, TagValueKey
+
+
+class PromptGenerationPluginInstance(PluginInstance):
+    """An instance of a configured prompt completion service such as GPT-3.
+
+    The `generate` method synchronously invokes the prompt against a set of variables that parameterize it.
+    The return value is a single string.
+
+    Example Usage:
+       llm = Steamship.use('prompt-generation-default', config={ "temperature": 0.9 })
+       PROMPT = "Greet {name} as if he were a {relation}."
+       greeting = llm.generate(PROMPT, {"name": "Ted", "relation": "old friend"})
+    """
+
+    def generate(self, prompt: str, variables: Optional[Dict] = None) -> str:
+        """Complete the provided prompt, interpolating any variables."""
+
+        # Interpolate the prompt with Python formatting semantics. If no variables provided, supply an empty dict.
+        try:
+            prompt_text = prompt.format(**(variables or {}))
+        except KeyError as e:
+            raise SteamshipError(
+                message="Some variables in the prompt template were not provided.", error=e
+            )
+
+        # Each prompt will be stored in Steamship as a File. The generated text will be
+        # associated with the prompt File via Steamship tags. This enables later retrieval
+        # of prompts and their results (which may be used in subsequent operations, etc.).
+        #
+        # To learn about the Steamship data model, please consult our the docs:
+        # https://docs.steamship.com/workspaces/data_model/index.html
+        file = File.create(self.client, blocks=[Block.CreateRequest(text=prompt_text)])
+
+        # This requests generation from the parameterized prompt. Tagging with our prompt generator
+        # plugin will result in a new tag that contains the generated output.
+        # We `wait()` because generation of text is done asynchronously and may take a few moments
+        # (somewhat depending on the complexity of your prompt).
+        tag_task = file.tag(plugin_instance=self.handle)
+
+        # Prompt completion happens asynchronously and remotely. We await the results here.
+        tag_task.wait()
+
+        # Here, we iterate through the content blocks associated with a file
+        # as well as any tags on that content to find the generated text.
+        #
+        # The Steamship data model provides flexible content organization,
+        # storage, and lookup. Read more about the data model via:
+        # https://docs.steamship.com/workspaces/data_model/index.html
+        for text_block in tag_task.output.file.blocks:
+            for block_tag in text_block.tags:
+                if block_tag.kind == TagKind.GENERATION:
+                    return self._clean_output(block_tag.value[TagValueKey.STRING_VALUE])
```

Comment:
If this isn't going to be in user code, I think there should be a flag on generate (default true) on whether to do this.  It's not clear to me it's the right behavior for all applications.

Reply:
Ooh. Good point.